geopandas/geodataframe.py

```diff
@@ -305,11 +313,17 @@ def set_geometry(self, col, drop=False, inplace=False, crs=None):
         else:
             frame = self.copy()
 
-        to_remove = None
```

Comment:
I thought I already left this comment but I guess not. I elected to get rid of the common path for `to_remove` since I don't think it makes this more readable - expecially now we would have to check `if geo_column_name in frame.columns` before calling del (but technically only required for the second case)